docs/conf.py

```diff
@@ -511,25 +511,45 @@ def _get_commit_refs(self, commit_sha):
         """Return a list of refs for the given commit sha"""
         return [ref for ref in self._repo.refs.keys() if self._repo.refs[ref] == commit_sha]
 
-    def _get_report_max_version(self):
-        # type: () -> Optional[Version]
-        """Determine the max cutoff version to report for HEAD"""
-        for entry in self._repo.get_walker():
+    def _get_report_max_version(self, max_commits=50):
+        # type: (int) -> Optional[Version]
+        """Determine the max cutoff version to report for HEAD
+
+        :param max_commits: Max number of commits to traverse looking for the latest origin/tag ref
+        """
+        for i, entry in enumerate(self._repo.get_walker()):
+            # If we have reached the max number of commits we want to look at without finding anything, return
+            # DEV: We should be able to find a proper ref within the first few, but if we do not then
+            #      it is better to return `None` and generate the notes for all versions, than to keep going backwards
+            #      looking for any ref that matches
+            if i >= max_commits:
+                return None
```

Comment:
```suggestion
        for entry, _ in zip(self._repo.get_walker(), range(max_commits)):
            # If we have reached the max number of commits we want to look at without finding anything, return
            # DEV: We should be able to find a proper ref within the first few, but if we do not then
            #      it is better to return `None` and generate the notes for all versions, than to keep going backwards
            #      looking for any ref that matches
```
but a bit obscure at first sight perhaps?

Reply:
/shrug, no strong opinion here